src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -1,14 +1,168 @@
+
+          //SCATTER
+        case s: Ast if s.getName.equals(AstNodeName.Scatter) =>
+          // Need to create the Scatter before generating its children to have the correct index layout order
+          val scatter = Scatter(s, scatterCount)
+          scatterCount += 1
+          val scatterChildren = getChildrenList(s) flatMap (generateScopeTree(_, namespaces, tasks, wdlSyntaxErrorFormatter))
+          scatter.setChildren(scatterChildren)
+          Some(scatter)
+
+        /* Cases need to be added here to support new scopes */
+        case _ => None
+      }
+    }
+  }
+
+  /**
+   * Generate a Scope Tree from an Ast Node.
+   * Node must match a Scope (currently: Workflow, Scatter or Scope)
+   * @param node node to build tree from
+   * @param namespaces namespaces used
+   * @param tasks tasks
+   * @param wdlSyntaxErrorFormatter wdl formatter
+   * @throws UnsupportedOperationException if node does not match any supported Scopes
+   * @return generated Scope Tree.
+   */
+  def generateTree(node: AstNode, namespaces: Seq[WdlNamespace], tasks: Seq[Task], wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Scope = {
+    new ScopeTreeGenerator().generateScopeTree(node, namespaces, tasks, wdlSyntaxErrorFormatter)
+    .getOrElse(throw new UnsupportedOperationException("Input node cannot be instantiated as a Scope"))
+  }
+  
+  /**
+   * Helper method to collect Calls from a Seq of Scopes. 
+   * /!\ NON-RECURSIVE.
+   * @param scopes scopes to loop through
+   * @return Scopes instances that are Calls
+   */
+  def collectCalls(scopes: Seq[Scope]): Seq[Call] = {
+    scopes collect {case s: Call => s}
+  }
+
+  /**
+   * Recursively collect all Calls from the given scope.
+   * @param scope scope to gather Calls from
+   * @return all Calls inside the scope
+   */
+  def collectAllCallsRec(scope: Scope): Seq[Call] = {
+    collectCalls(Seq(scope)) ++ (scope.children flatMap collectAllCallsRec)
+  }
+
+  /**
+   * Helper method to collect Scatters from a Seq of Scopes. 
+   * /!\ NON-RECURSIVE.
+   * @param scopes scopes to loop through
+   * @return Scopes instances that are Scatters
+   */
+  def collectScatters(scopes: Seq[Scope]): Seq[Scatter] = {
+    scopes collect {case s: Scatter => s}
+  }
+
+  /**
+   * Recursively collect all Scatters from the given scope.
+   * @param scope scope to gather Scatters from
+   * @return all Scatters inside the scope
+   */
+  def collectAllScattersRec(scope: Scope): Seq[Scatter] = {
+    collectScatters(Seq(scope)) ++ (scope.children flatMap collectAllScattersRec)
+  }
+
+}
+
 // FIXME: It'd be nice to have a notion of a parented and a not-parented Scope, see FIXME in Call about this
 trait Scope {
+
   def name: String
 
-  def parent: Option[Scope]
+  private var _parent: Option[Scope] = None
+
+  def parent: Option[Scope] =_parent
+
+  private var _children: Seq[Scope] = Seq.empty[Scope]
+
+  def children: Seq[Scope] =_children
+
 
```

Comment:
It makes sense that we'd want some way to get the FQN of a Scatter object... though I'm not sure the best way to do this because I see why you did it this way.

Maybe an optional parameter on fullyQualifiedName?  Maybe a separate method?